backend_jobs/recognition_pipeline/providers/google_vision_api.py

```diff
@@ -0,0 +1,56 @@
+"""
+  Copyright 2020 Google LLC
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+"""
+
+from google.cloud import vision_v1
+from pipeline_lib.image_recognition_provider import ImageRecognitionProvider
+from filters.filter_by_format import FilterByFormat
+from filters.filter_by_resolution import FilterByResolution
+
+# pylint: disable=abstract-method
+class GoogleVisionAPI(ImageRecognitionProvider):
+    """ An implementation of the abstract class ImageRecognitionProvider
+    specified for Google Vision API.
+
+    """
+
+    def setup(self):
+        self.client = vision_v1.ImageAnnotatorClient()
+
+    # pylint: disable=arguments-differ
+    def process(self, element):
+        features = [ {"type_": vision_v1.Feature.Type.LABEL_DETECTION} ]
+        requests = []
+        results = []
+        docs = []
+        for doc in element[1]:
```

Comment:
how many docs max can we expect here? is there a risk we're going to send too many at once to the API?

Reply:
fixed this using a loop - takes a batch of max 2000.